drivers: regulator: API to get max regulator voltage

RafaelLaya opened the pull request:
There is already a regulator_common_get_min_voltage() API, to query the maximum supported voltage on a regulator. This change adds an analogous and symmetric API to query the maximum supported voltage

Commit:
drivers: regulator: API to get max regulator voltage

There is already a get_min_voltage() API, so let us add
an API to also query the max voltage

Signed-off-by: Rafael Laya <rafael_laya97@hotmail.com.com>

RafaelLaya commented:
@gmarull @danieldegrasse @aasinclair Fixed title and commit message (this is on the regulator API, not i2c, but for some reason my brain made me type i2c...) Hope all the automatic tags are correct?

gmarull reviewed (commented):
where is this needed/used? These are internal APIs

RafaelLaya commented:
> where is this needed/used? These are internal APIs

Good point. I've added two more commits, for a total of three:

1. The addition of get_max_voltage()
2. Adds unit-testing coverage for the new get_max_voltage() and for the pre-existing get_min_voltage()
3. Linter pass on the unit tests to make sure CI is happy

Now it's used in the tests

Regarding for where this is needed:

1. For symmetry since the get_min_voltage() internal API already exists
2. It is useful for code interfacing with regulators. I've some uses for it but it's not ready yet (if ever) to upstream. But in any case, this can still be a useful API for users interfacing with regulators at the low level

gmarull reviewed (changes requested):
please, introduce any formatting changes in a separate commit.

Commit:
drivers: regulator: add tests for get_max/min voltage

Adds unit tests to make sure get_max and get_min voltage
remain healthy

Signed-off-by: Rafael Laya <rafael_laya97@hotmail.com.com>

Commit:
drivers: regulator: Linter pass to unit test

Linter pass makes CI happy and keeps code compliant
with the stylistic guidelines of the project

Signed-off-by: Rafael Laya <rafael_laya97@hotmail.com.com>

RafaelLaya commented:
> please, introduce any formatting changes in a separate commit.

Good catch! I thought I had three commits. My bad, now I've actually split the changes into the three commits mentioned above
